workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogSession.kt

```diff
@@ -68,7 +74,18 @@ internal class DialogSession(
         }
 
         override fun dispatchKeyEvent(event: KeyEvent): Boolean {
-          return !allowEvents || realWindowCallback.dispatchKeyEvent(event)
+          // Consume all events if we've been told to do so.
+          if (!allowEvents) return true
+
+          // If there is an onBackPressed handler invoke it instead of allowing
+          // the normal machinery to call Dialog.onBackPressed.
+          if (event.isBackPress) {
+            holder.onBackPressed?.invoke()
+            return true
+          }
+
+          // Allow the usual call to Dialog.onBackPressed.
```

Comment:
Confused. This isn't allowing the usual call to `onBackPressed` as that is handled above no? Do you mean route the regular `dispatchKeyEvent` handling for all other keys?